jetty-server/src/test/java/org/eclipse/jetty/server/ErrorHandlerTest.java

```diff
@@ -233,6 +233,88 @@ public void test404HtmlAccept() throws Exception
         assertContent(response);
     }
 
+    @Test
+    public void test404PostHttp10() throws Exception
+    {
+        String rawResponse = connector.getResponse(
+            "POST / HTTP/1.0\r\n" +
+                "Host: Localhost\r\n" +
+                "Accept: text/html\r\n" +
+                "Content-Length: 10\r\n" +
+                "Connection: keep-alive\r\n" +
+                "\r\n" +
+                "0123456789");
+
+        HttpTester.Response response = HttpTester.parseResponse(rawResponse);
+
+        assertThat(response.getStatus(), is(404));
+        assertThat(response.getField(HttpHeader.CONTENT_LENGTH).getIntValue(), greaterThan(0));
+        assertThat(response.get(HttpHeader.CONTENT_TYPE), containsString("text/html;charset=ISO-8859-1"));
+        assertThat(response.get(HttpHeader.CONNECTION), is("keep-alive"));
+        assertContent(response);
+    }
+
+    @Test
+    public void test404PostHttp11() throws Exception
+    {
+        String rawResponse = connector.getResponse(
+            "POST / HTTP/1.1\r\n" +
+                "Host: Localhost\r\n" +
+                "Accept: text/html\r\n" +
+                "Content-Length: 10\r\n" +
+                "\r\n" +
+                "0123456789");
+
+        HttpTester.Response response = HttpTester.parseResponse(rawResponse);
+
+        assertThat(response.getStatus(), is(404));
+        assertThat(response.getField(HttpHeader.CONTENT_LENGTH).getIntValue(), greaterThan(0));
+        assertThat(response.get(HttpHeader.CONTENT_TYPE), containsString("text/html;charset=ISO-8859-1"));
+        assertThat(response.getField(HttpHeader.CONNECTION), nullValue());
+        assertContent(response);
+    }
```

Comment:
One more test with HTTP/1.1 with `Connection: keep-alive`?

Reply:
Can we also add a test with gzip content? Can we assert that we did not ungzip it?

Reply:
@sbordet I've added `keep-alive` to the existing 1.1 test as it doesn't change behaviour, but shows that it is correctly ignored.
@joakime can I get you to do that gzip test?